addons/http_routing/models/ir_http.py

```diff
@@ -185,15 +185,15 @@ def url_for(url_from, lang_code=None, no_rewrite=False):
         :param no_rewrite: don't try to match route with website.rewrite.
     '''
     new_url = False
-
+    rewrite = not no_rewrite
     # don't try to match route if we know that no rewrite has been loaded.
     routing = getattr(request, 'website_routing', None)  # not modular, but not overridable
-    if not getattr(request.env['ir.http'], '_rewrite_len', {}).get(routing):
-        no_rewrite = True
+    if not request.env['ir.http']._rewrite_len(routing):
```

Comment:
It looks like a miss copy in the commit reference. But yes the revert done by moc was done because we noticed that the refactoring was not correct, as I explained in the previous comment about the routing map cls is registry specific and thus the behavior is not the same.

This change is not introducing the same issue since the ormcache is database specific, or did I misunderstand your commit? 

Anyway, good catch for the missing _rewrite_len I will fix this by adding it in http_routting for now! Maybe that we could also rename it in `has_rewrites` or something like that?

Reply:
no, was just about getattr only (that we already try to remove :innocent: ).
So yes just re-add it (or move function in http_routing and overrides in website), and good time to rename it if you prefer